test/SparqlAntlrParserTest.cpp

```diff
@@ -20,14 +20,14 @@ using namespace sparqlParserHelpers;
 
 template <typename T>
 void testNumericLiteral(const std::string& input, T target) {
-  ParserAndVisitor p(input);
-  auto literalContext = p.parser_.numericLiteral();
-  auto result = p.visitor_.visitNumericLiteral(literalContext).as<T>();
+  auto result = sparqlParserHelpers::parseNumericLiteral(input);
+  ASSERT_EQ(result.remainingText_.size(), 0);
+  auto value = get<T>(result.resultOfParse_);
 
   if constexpr (std::is_floating_point_v<T>) {
-    ASSERT_FLOAT_EQ(target, result);
+    ASSERT_FLOAT_EQ(target, value);
```

Comment:
I think even "ASSERT_DOUBLE_EQ" should pass here, as we have the full double precision in this case.